simde/x86/aes.h

```diff
@@ -0,0 +1,603 @@
+      s++;
+    }
+  }
+}
+
+/*
+ * Transformation in the Inverse Cipher that is the inverse of 
+ * ShiftRows().
+ */
+SIMDE_FUNCTION_ATTRIBUTES
+void softaes_inv_shift_rows(uint8_t *state) {
+
+  int Nb = softaes_Nb;
+  uint8_t i, k, s, tmp;
+
+  for (i = 1; i < 4; i++) {
+    s = 0;
+    while (s < i) {
+      tmp = state[Nb*i+Nb-1];
+
+      for (k = Nb-1; k > 0; k--) {
+        state[Nb*i+k] = state[Nb*i+k-1];
+      }
+
+      state[Nb*i+0] = tmp;
+      s++;
+    }
+  }
+}
+
+/*
+ * Transformation in the Cipher that processes the State using a non
+ * linear byte substitution table (S-box) that operates on each of the 
+ * State bytes independently. 
+ */
+SIMDE_FUNCTION_ATTRIBUTES
+void softaes_sub_bytes(uint8_t *state) {
+
+  int Nb = softaes_Nb;
+  uint8_t i, j;
+
+  for (i = 0; i < 4; i++) {
+    for (j = 0; j < Nb; j++) {
+      // s_box row: yyyy ----
+      // s_box col: ---- xxxx
+      // s_box[16*(yyyy) + xxxx] == s_box[yyyyxxxx]
+      state[Nb*i+j] = softaes_s_box[state[Nb*i+j]];
+    }
+  }
+}
+
+/*
+ * Transformation in the Inverse Cipher that is the inverse of 
+ * SubBytes().
+ */
+SIMDE_FUNCTION_ATTRIBUTES
+void softaes_inv_sub_bytes(uint8_t *state) {
+
+  int Nb = softaes_Nb;
+  uint8_t i, j;
+
+  for (i = 0; i < 4; i++) {
+    for (j = 0; j < Nb; j++) {
+      state[Nb*i+j] = softaes_inv_s_box[state[Nb*i+j]];
+    }
+  }
+}
+
+/*
+ * Performs the AES cipher operation
+ */
+SIMDE_FUNCTION_ATTRIBUTES
+void softaes_enc(uint8_t *in, uint8_t *out, uint8_t *w, int is_last) {
+
+  int Nb = softaes_Nb;
+  uint8_t state[4*Nb];
+  uint8_t r = 0, i, j;
+
+  for (i = 0; i < 4; i++) {
+    for (j = 0; j < Nb; j++) {
+      state[Nb*i+j] = in[i+4*j];
+    }
+  }
+
+  softaes_sub_bytes(state);
+  softaes_shift_rows(state);
+
+  if (!is_last)
+    softaes_mix_columns(state);
+
+  softaes_add_round_key(state, w, r);
+
+  for (i = 0; i < 4; i++) {
+    for (j = 0; j < Nb; j++) {
+      out[i+4*j] = state[Nb*i+j];
+    }
+  }
+}
+
+/*
+ * Performs the AES inverse cipher operation
+ */
+SIMDE_FUNCTION_ATTRIBUTES
+void softaes_dec(uint8_t *in, uint8_t *out, uint8_t *w, int is_last) {
+
+  int Nb = softaes_Nb;
+  uint8_t state[4*Nb];
+  uint8_t r = 0, i, j;
+
+  for (i = 0; i < 4; i++) {
+    for (j = 0; j < Nb; j++) {
+      state[Nb*i+j] = in[i+4*j];
+    }
+  }
+
+  softaes_inv_shift_rows(state);
+  softaes_inv_sub_bytes(state);
+
+  if (!is_last)
+    softaes_inv_mix_columns(state);
+
+  softaes_add_round_key(state, w, r);
+
+  for (i = 0; i < 4; i++) {
+    for (j = 0; j < Nb; j++) {
+      out[i+4*j] = state[Nb*i+j];
+    }
+  }
+}
+
+SIMDE_FUNCTION_ATTRIBUTES
+simde__m128i simde_mm_aesenc_si128(simde__m128i a, simde__m128i round_key) {
+  #if defined(SIMDE_X86_AES_NATIVE)
+    return _mm_aesenc_si128(a, round_key);
+  #else
+    simde__m128i result;
+    softaes_enc((uint8_t *) &a, (uint8_t *) &result, (uint8_t *) &round_key, 0);
+    return result;
+  #endif
+}
+#if defined(SIMDE_X86_AES_ENABLE_NATIVE_ALIASES)
```

Comment:
Please define `SIMDE_X86_AES_ENABLE_NATIVE_ALIASES` in `simde/simde-features.h` (around line 568)